features/netsocket/UDPSocket.cpp

```diff
@@ -128,7 +144,9 @@ nsapi_size_or_error_t UDPSocket::recvfrom(SocketAddress *address, void *buffer,
 
 void UDPSocket::event()
 {
+#ifdef MBED_CONF_RTOS_PRESENT
     _event_flag.set(READ_FLAG|WRITE_FLAG);
+#endif
```

Comment:
I wonder if this would be better as a PlatformSemaphore (and PlatformEventFlags?).